### ISSUE TITLE
Salesforce: "Social Network Login Failure"

### ISSUE THREAD

**Issue by sunweiyang** [pennersr/django-allauth#2732]
I have a Django backend with django-allauth, and I'm trying to add Salesforce as a login method.

I've followed the [instructions here](https://django-allauth.readthedocs.io/en/latest/providers.html#salesforce):
* Created a Salesforce Connected App with id and openid scopes (along with some others), and set the callback URL to https://www.mywebdomain.com/accounts/salesforce/login/callback/ (not my real domain, obviously)
* Created a `SocialApplication` in Django with client ID, secret, and login URL in the "Key" field (https://login.salesforce.com/)
* Included allauth.socialaccount.providers.salesforce in INSTALLED_APPS

After entering Salesforce login credentials, the Salesforce system redirected to the callback URL, resulting in a page that contains the following text:
```
Social Network Login Failure
An error occurred while attempting to login via your social network account.
```

The URL in the address bar on this page looks like this:
```
https://www.mywebdomain.com/accounts/salesforce/login/callback/#access_token=00D3t000004QWRm%21ARwAQPfHWiM6jdB43dlyW6qjEw._34mjzGi_Jv6YCXp0QssT.9F9lCge5_YaH8gqTy3Od6SywCs8X9zOGv145SyviBVeGdn0&instance_url=https%3A%2F%2Fna123.salesforce.com&id=https%3A%2F%2Flogin.salesforce.com%2Fid%2F00D3t000004QWRmEAO%2F0053t000008QBetAAG&issued_at=1606802917608&signature=KvxAX0WBCFQYY%2BO25id9%2FXxpbh2q2d2vWdQ%2FFV5FCBw%3D&state=jsforce0.popup.c0ockgct29g&scope=id+api+web+refresh_token+openid&token_type=Bearer
```

I tried to debug and print the error (`auth_error.code` and `auth_error.exception`), but they were empty.

Am I missing something, or is the Salesforce provider experiencing an issue as currently implemented?

**Comment by sunweiyang** [comment]
I figured it out and got it working: when posting to my dj-rest-auth Salesforce API endpoint, I was only including `access_token` in my POST body. I actually need both `access_token` and `key`, where `key` is the Salesforce login URL ("https://login.salesforce.com").

It was indeed in the django-allauth instructions for Salesforce, but I misinterpreted the wording. I now know that it says to require both `access_token` and `key` in the POST body.
